e2e/transactions/transactions_test.go

```diff
@@ -784,15 +784,17 @@ func (s *TransactionsTestSuite) TestEvictionOfQueuedTransactions() {
 
 	txQueue := s.Backend.TxQueueManager().TransactionQueue()
 
-	var i = 0
+	var i = int32(0)
 	txIDs := [txqueue.DefaultTxQueueCap + 5 + 10]common.QueuedTxID{}
 	txQueueMutex := sync.RWMutex{}
 	s.Backend.TxQueueManager().SetTransactionQueueHandler(func(queuedTx *common.QueuedTx) {
```

Comment:
Correct me if I'm wrong, but the race (one of them at least) here happens because concurrent calls to `Enqueue()` and `SetEnqueueHandler()` in `geth/txqueue/txqueue.go` are not guarded. Thus it makes sense to guard them in the `txqueue` implementation, rather than fix shared state updates (`i`) in tests.

Reply:
Nope, actually. This data race is only about `i`, `Enqueue` reads and changes `i` but in test we read `i` too:
```
log.Info(fmt.Sprintf("Number of transactions queued: %d. Queue size (shouldn't be more than %d): %d", i, txqueue.DefaultTxQueueCap, txQueue.Count()))
```

Reply:
@divan Is it ok?